app/controllers/foreman_deployments/api/v2/deployments_controller.rb

```diff
@@ -0,0 +1,73 @@
+module ForemanDeployments
+  module Api
+    module V2
+      class DeploymentsController < BaseController
+        before_filter :find_resource, :only => [:show, :configure, :configuration, :run]
+        before_filter :parse_stack, :only => [:show, :configure, :configuration, :run]
+
+        def_param_group :deployment do
+          param :deployment, Hash, :required => true, :action_aware => true do
+            param :name, String, :required => true, :desc => N_('Name for the deployment')
+            param :stack_id, :identifier, :required => true, :desc => N_('Id of the stack to deploy')
+          end
+        end
+
+        api :POST, '/deployments/', N_('Create a deployment')
+        param_group :deployment, :as => :create
+        def create
+          deployment_params = params[:deployment]
+
+          stack = ForemanDeployments::Stack.find(deployment_params.delete(:stack_id)) if deployment_params[:stack_id]
+          deployment_params[:configuration] = ForemanDeployments::Configuration.new(:stack => stack)
+
+          @deployment = ForemanDeployments::Deployment.new(deployment_params)
+          process_response @deployment.save
+        end
+
+        api :GET, '/deployments/', N_('List deployments')
+        def index
+          @deployments = resource_scope
+        end
+
+        api :PUT, '/deployments/:id/configuration/', N_('Configure a deployment')
+        param :id, :identifier, :required => true
+        param :values, Hash, :required => true,
+                             :desc => N_('Hash with configuration for tasks (Format: task ID => configuration values).')
+        def configuration
+          configuration_update = ForemanDeployments::Configuration.new(:values => params[:values])
+
+          ForemanDeployments::Config::LoadVisitor.load(@stack_definition, @deployment.configuration)
+          ForemanDeployments::Config::LoadVisitor.load(@stack_definition, configuration_update)
+          ForemanDeployments::Config::SaveVisitor.save(@stack_definition, @deployment.configuration)
```

Comment:
Probably this can be moved to the `Configuration` model along with the previous `LoadVisitor` line:

``` Ruby
module ForemanDeployments
  class Configuration < ActiveRecord::Base

     # ...

    def merge_configuration(configuration_update)
      return unless deployment

      stack = deployment.parsed_stack # see a comment on parse_stack method
      ForemanDeployments::Config::LoadVisitor.load(stack, configuration_update)
      ForemanDeployments::Config::SaveVisitor.save(stack, self)
    end
  end
```